.github/workflows/upgrade_downgrade_test_semi_sync.yml

```diff
@@ -0,0 +1,164 @@
+name: Semi Sync Upgrade Downgrade Testing
+on:
+  push:
+  pull_request:
+
+concurrency:
+  group: format('{0}-{1}', ${{ github.ref }}, 'Semi Sync Upgrade Downgrade Testing')
```

Comment:
Shouldn't we also have that test on `main` to avoid future changes that are incompatible?

Reply:
Good question. I thought about this too, but it seemed like we won't be making any more changes to semi-sync config on main.

Reply:
I can add a workflow to main, but it isn't all that necessary in my opinion. But we can always add a test similar to this on main too.

Reply:
I feel like we probably should, we already have the code for the test, running a new workflow won't be very expensive, and will only increase our test coverage.